cassandraprotocol/client/handshake.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"fmt"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol/frame"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol/message"
+)
+
+func StartupRequest(c *CqlConnection, version cassandraprotocol.ProtocolVersion, streamId int16) *frame.Frame {
+	var startup *message.Startup
+	if c.codec.CompressionAlgorithm() == "NONE" {
+		startup = message.NewStartup()
+	} else {
+		startup = message.NewStartup(message.WithCompression(c.codec.CompressionAlgorithm()))
+	}
+	request, _ := frame.NewRequestFrame(version, streamId, false, nil, startup)
+	return request
+}
+
+func Handshake(client *CqlConnection, version cassandraprotocol.ProtocolVersion, streamId int16) error {
+	if err := client.Send(StartupRequest(client, version, streamId)); err != nil {
+		return err
+	} else if response, err := client.Receive(); err != nil {
+		return err
+	} else if _, ok := response.Body.Message.(*message.Ready); !ok {
+		return fmt.Errorf("expected READY, got %T", response.Body.Message)
+	} else {
+		return nil
+	}
+}
+
+func HandshakeAuth(client *CqlConnection, version cassandraprotocol.ProtocolVersion, streamId int16, username string, password string) error {
```

Comment:
Handshake procedure with authentication enabled. Tested against C* 3.11 and DSE 6.0.